internal/core/domain/stats_repository.go

```diff
@@ -0,0 +1,23 @@
+package domain
+
+import "context"
+
+type Page struct {
+	Number int
+	Size   int
+}
+
+type StatsRepository interface {
```

Comment:
I think it's better to have 2 distinct repos for withdrawals and deposits. Considering these as 2 different tables (for relational DBs) or buckets (for key-value DBs), or in general DB files, IMO it would be better to stick with the concept "one repo - one db file". This way, in a certain way we are enforcing at the domain level that things in the concrete DB must be separated in order to grant exlusive lock on only a single file (wheter it contains a table or a bucket).

This would make the operator able to do a withdrawal while depositing some funds. If withdrawals and deposits go to the same file that would not be possible so to speak.

Reply:
agreed @sekulicd

Reply:
Done